### PR TITLE
Release Candidate V0.28.2

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** Version 0.28.1 has a Python versioning bug while building Linux wheels.

**Description of the Change:** Fix Python module versioning for Linux wheels.

**Benefits:** Wheels will be built properly.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by multiphaseCFD** [comment]
Nice work @AmintorDusko! Thanks. Just leave a minor question there.

File: setup.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def build_extension(self, ext: CMakeExtension):
         # Set Python_EXECUTABLE instead if you use PYBIND11_FINDPYTHON
         configure_args = [
             f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}",
-            f"-DPython_EXECUTABLE={sys.executable}",
+            f"-DPython_EXECUTABLE={sys.executable}", # (Windows)
+            f"-DPYTHON_EXECUTABLE={sys.executable}", # (Ubuntu)
```
Comment (multiphaseCFD) [inline setup.py:57]: What about macOS?

Comment (AmintorDusko) [reply setup.py:57]: macOS works fine with both.

**Review by multiphaseCFD — APPROVED** [review]

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,7 +1,8 @@
 ninja
 flaky
 numpy
-git+https://github.com/PennyLaneAI/pennylane.git@master
+# git+https://github.com/PennyLaneAI/pennylane.git@master
```
[the remark this answers is not in the document]
Comment (AmintorDusko) [reply requirements-dev.txt:4]: Yes. To build the wheels, after that I will change back.
I had some comments here pointing this out but I forgot to submit it.

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -1,16 +1,15 @@
-# Release 0.29.0-dev
+# Release 0.28.2
 
-### New features since last release
-
-### Breaking changes
+### Bug fixes
 
-### Improvements
+* Fix Pybind11 module versioning and locations for Linux wheels.
```
Comment (mlxd) [inline .github/CHANGELOG.md:5]:
```suggestion
* Fix Python module versioning for Linux wheels.
```

File: .github/workflows/wheel_macos_x86_64.yml [hunk 4]
```diff
@@ -186,7 +186,7 @@ jobs:
           regex: '.*[0-9]+.[0-9]+.[0-9]+[-_]?rc[0-9]+'
 
       - uses: actions/upload-artifact@v2
-        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
+        # if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```
[the remark this answers is not in the document]
Comment (mlxd) [reply .github/workflows/wheel_macos_x86_64.yml:189]: Feel free to undo this change. Wheels look good.

File: .github/workflows/wheel_win_x86_64.yml [hunk 5]
```diff
@@ -189,7 +189,7 @@ jobs:
           regex: '.*[0-9]+.[0-9]+.[0-9]+[-_]?rc[0-9]+'
 
       - uses: actions/upload-artifact@v2
-        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
+        # if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```
[the remark this answers is not in the document]
Comment (mlxd) [reply .github/workflows/wheel_win_x86_64.yml:192]: Same here

File: requirements-dev.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ninja
 flaky
 numpy
-git+https://github.com/PennyLaneAI/pennylane.git@master
+pennylane~=0.28
```
Comment (mlxd) [inline requirements-dev.txt:4]: This should also be back to PennyLane master?

**Review by mlxd — COMMENTED** [review]
Thanks Amintor.

File: requirements-dev.txt (hunk 6 above)
Comment (AmintorDusko) [reply requirements-dev.txt:4]: Yes. But just after you fork the release branch. Right?

Comment (mlxd) [reply requirements-dev.txt:4]: Nope. The development requirements are always mainline with PennyLane, as they arent packages in the wheel

Comment (AmintorDusko) [reply requirements-dev.txt]: Just changed it back.

**Comment by AmintorDusko** [comment]
This branch was forked to v0.28.2_release and tagged for release.
Now, this PR will be updated for merging into the master.

**Review by multiphaseCFD — APPROVED** [review]
Nice work, thanks!

**Pull request merged**